docs/docs/QuickAddAPI.md

```
@@ -610,6 +610,135 @@ const result = await quickAddApi.ai.chunkedPrompt(
+  Unsupported keywords (`pattern`, `additionalProperties`, `$ref`, `format`, …) are rejected
+  at registration so a provider can't silently drop a constraint.
+- `execute(input, ctx)` runs your code with the model-chosen `input` (validated against the
+  schema first). Return a string (used verbatim) or any JSON-serialisable value.
+- `needsApproval` (boolean or `(args) => boolean`) asks before running. `readOnly: true` marks a
+  tool that only reads, so it auto-runs under the default confirmation setting.
+
+> **Confirmation needs an interactive Obsidian session.** A tool that asks for approval opens a
+> modal and waits for it. For unattended automation (e.g. driving QuickAdd from the CLI), give the
+> agent only `readOnly` tools, or set **Confirm AI tool calls** to *Never* and gate each tool with
+> its own `needsApproval` — otherwise the run blocks on a dialog no one can answer.
+
```

markdownlint-cli2 (0.22.1)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)